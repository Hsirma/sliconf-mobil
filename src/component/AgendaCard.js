import React, { Component } from 'react';
import { View, Text, StyleSheet,StatusBar,TouchableOpacity,Image,Dimensions,Platform } from 'react-native';
import { Container, Header, Left, Body, Right, Button, Icon, Title ,Content,CardItem,Thumbnail,Card,List,Item,Footer,FooterTab,Picker, Form, Item as FormItem} from 'native-base';
import PropTypes from 'prop-types';


export default class AgendaCard extends Component {

    state={
        isExist:true,
        isClicked:this.props.isClicked
    }

    /**
     * Konusmanin leveline gore renk dondurur.
     * @param level
     * @returns {*}
     */
    getColorByLevel(level){
        switch(level){
            case 1:
                return '#29B673';
                break;
            case 2:
                return '#FBB041';
                break;
            case 3:
                return '#EE5E5F';
                break;
            default :
                return '#ffffff';
        }
    }
    setButtonIcon(){
        this.setState(
            {
                isClicked:!this.state.isClicked
            }
        )
    }

    /**
     * Cardların ustundeki butonların ne olacagini ayarlar.
     * @param obj
     */
    handleClick(obj){
        this.setButtonIcon();
        let tempObject=obj;
        if(this.isSameData(obj)){
            this.props.onPressDeleteButton(tempObject);
        }else{
            this.props.onPressAddButton(tempObject);
        }

    }

    /**
     * Card verisi sectiklerim listesinde varsa true yoksa false dondurur.
     * @param data
     * @returns {boolean}
     */
    isSameData(data){
        let choosedOne=this.props.choosedEvents;
        let obj = choosedOne.find((data2)=>data===data2)
        if(obj!== undefined)
            return true
        return false
    }
    render() {
        const item =this.props.item;
        if(this.props.isEmpty){
            return(
                <View  style={styles.container}>

                    <Image source={require('../../images/emptyCard.png')} style={{width:200,height:100}}/>
                </View>
            )
        }else{
            return (
                <TouchableOpacity  style={styles.container} onPress={this.props.onPress}>
                    <View  style={[styles.cardLine,{borderColor:this.getColorByLevel(item.level)}]}/>
                    <View style={styles.detailField}>
                        <Text style={styles.topic}>{item.topic} </Text>
                        <Text style={styles.speaker}>{item.speaker}</Text>
                    </View>
                    <View style={styles.actionField}>
                        <Thumbnail source={require('../../images/hi.png')}   />
                        <TouchableOpacity
                            onPress={() => this.handleClick(item)} >
                            <View style={styles.buttonField}>
                                <Icon name={this.state.isClicked? 'ios-checkmark':'ios-add'}
                                      style={{alignSelf:'center'}}/>
                            </View>
                        </TouchableOpacity>
                    </View>
                </TouchableOpacity>
            );
        }

    }
}
const styles = StyleSheet.create({
    container: {
        flexDirection:'row',
        width:200,
        height:100,
        margin:5,
        backgroundColor:'#fff',
        borderWidth:1,
        borderRadius:15,
        borderColor:'#F1F2F2',
    },
    cardLineEmpty:{
        borderColor:'#000',
        borderWidth:1,
        margin:10
    },
    cardLine:{
        borderWidth:1,
        margin:10,
        marginTop:0,
        marginBottom:0
    },
    topic: {
        fontSize:10,
        textAlign:'left',
        textAlignVertical:'center',
        color:'#000000',
        margin:5
    },
    speaker:{
        fontSize:8,
        textAlign:'left',
        margin:5
    },
    detailField:{
        flex:0.7,
        justifyContent:'space-between'
    },
    actionField:{
        flex:0.3,
        marginRight:10,
        justifyContent:'space-around'
    },
    buttonField:{
        backgroundColor:'#F1F2F2',
        width:30,
        height:30,
        borderRadius:100,
        alignSelf:'flex-end'
    }

})