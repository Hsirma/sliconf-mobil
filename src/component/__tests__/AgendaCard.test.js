import React from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import {Icon} from 'native-base';
import {shallow} from 'enzyme';
import {expect} from 'chai';
import sinon from "sinon";
import AgendaCard from "../AgendaCard";
import renderer from 'react-test-renderer';

const mockData = {
    "key": "a102",
    "time": '9:30',
    "topic": "CI/CD of blockchain smart1 contracts using Java and eDuke",
    "topicDetail": "Blockchain is a hot topic especially the smart contract feature. Smart contracts allow to customize the rules applicable to digital assets deployed on a blockchain. On the Ethereum blockchain, Solidity is the usual programming language used to develop smart contract. With the use of eDuke, a Java framework allowing easy interactions with the Ethereum blockchain, we will show how to continuously deploy and test smart contracts and 'oracle' code using JUnit, Jenkins and Maven.",
    "level": 2,
    "tags": [
        "Java",
        "JVM",
        "Security",
    ],
    "room": "Big Saloon",
    "speaker": "Frédéric Hubin",
    "star": 4.5,
    "date": '1525510807',
};
const speaker = {
    "id": "string",
    "name": "string",
    "profilePicture": ""

};
describe('<AgendaCard />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <AgendaCard item={mockData} speaker={speaker} isClicked={false}/>
        );
    });
    it('it should isClicked false', () => {
        const wrapper = shallow(<AgendaCard item={mockData} speaker={speaker} isClicked={false}/>);
        expect(wrapper.state().isClicked).to.not.equal(undefined)
    });
    it('it should render 5 View component', () => {
        const wrapper = shallow(<AgendaCard item={mockData} speaker={speaker}/>);
        expect(wrapper.find(View)).to.have.length(5);
    });

    it('it should render 3 text components', () => {
        const wrapper = shallow(<AgendaCard item={mockData} speaker={speaker}/>);
        expect(wrapper.find(Text)).to.have.length(3);
    });
    it('it should render 2 TouchableOpacity components', () => {
        const wrapper = shallow(<AgendaCard item={mockData} speaker={speaker}/>);
        expect(wrapper.find(TouchableOpacity)).to.have.length(2);
    });
    it('it should render 1 Icon components', () => {
        const wrapper = shallow(<AgendaCard item={mockData} speaker={speaker}/>);
        expect(wrapper.find(Icon)).to.have.length(1);
    });
    it("it should handle button presses", () => {
        const onPress = sinon.spy();
        const button = shallow(<AgendaCard item={mockData} speaker={speaker} onPress={onPress}/>);
        button.simulate('press');
        expect(onPress.calledOnce).to.equal(true);
    });
    it('calls getColorByLevel', () => {
        const wrapper = shallow(<AgendaCard item={mockData} speaker={speaker}/>);
        expect(wrapper.instance().getColorByLevel(2)).to.equal('#EE5E5F')
    });
});

describe('if <AgendaCard /> is empty', () => {
    it('it should render 1 View component', () => {
        const wrapper = shallow(<AgendaCard isEmpty={true}/>);
        expect(wrapper.find(View)).to.have.length(1);
    });
    it('it should render 1 Image components', () => {
        const wrapper = shallow(<AgendaCard isEmpty={true}/>);
        expect(wrapper.find(Image)).to.have.length(1);
    });
});

