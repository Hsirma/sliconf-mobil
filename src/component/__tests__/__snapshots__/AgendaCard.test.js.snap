// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AgendaCard /> with mockData Renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#D1D3D4",
      "borderRadius": 15,
      "borderWidth": 0.5,
      "flexDirection": "row",
      "height": 120,
      "margin": 5,
      "opacity": 1,
      "width": 220,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "borderWidth": 1,
          "margin": 10,
          "marginBottom": 0,
          "marginTop": 0,
        },
        Object {
          "borderColor": "#FBB041",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 0.7,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 14.142857142857142,
          "margin": 5,
          "textAlign": "left",
          "textAlignVertical": "center",
        }
      }
    >
      A mobile-first approach to 3DSecure today
       
    </Text>
    <View
      style={
        Object {
          "marginBottom": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#414042",
            "fontFamily": "Montserrat-Light",
            "fontSize": 12.571428571428571,
            "margin": 5,
            "marginBottom": 0,
            "textAlign": "left",
          }
        }
      >
        string
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#6D6E71",
            "fontFamily": "Montserrat-Light",
            "fontSize": 11,
            "margin": 5,
            "marginBottom": 0,
            "marginTop": 0,
            "textAlign": "left",
          }
        }
      >
        Tags: 
        Security
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.3,
        "justifyContent": "space-around",
        "marginRight": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "borderRadius": 28,
          "height": 56,
          "width": 56,
        }
      }
      virtual={undefined}
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignSelf": "flex-end",
            "backgroundColor": "#F1F2F2",
            "borderRadius": 100,
            "height": 30,
            "width": 30,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "alignSelf": "center",
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          virtual={undefined}
        >
          ï„‚
        </Text>
      </View>
    </View>
  </View>
</View>
`;
